src/instruction.rs

```diff
@@ -0,0 +1,185 @@
+//! Program instructions
+
+use solana_program::clock::UnixTimestamp;
+
+/// Instructions supported by the staking program
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum StakeInstruction {
+    /// Creates Stake config account which controls staking parameters
+    ///
+    /// 0. `[writable]` Stake config account
+    /// 1. `[]` Slash authority, may be a PDA for a governance program
+    /// 2. `[]` Config authority, may be a PDA for a governance program
+    /// 3. `[]` Stake Token Mint
+    /// 4. `[]` Stake Token Vault, to hold all staked tokens.
+    ///   Must be empty and owned by PDA with seeds `['token-owner', stake_config]`
+    InitializeConfig {
+        cooldown_time: UnixTimestamp,
+        max_deactivation_basis_points: u16,
+    },
+
+    /// Initializes stake account data for a validator.
+    ///
+    /// NOTE: Anybody can create the stake account for a validator. The authority
+    /// defaults to the validator withdraw authority.
```

Comment:
"Defaults" or "is always set to"?

Reply:
If you create a stake account for someone else, the authority will default to the validator withdraw authority. The validator may then set it to any other key. So it won't *always* be set to the withdraw authority. I can clarify, "For new stake accounts, the authority defaults..." -- what do you think?

Reply:
Would the validator be using the `SetAuthority` instruction to do this reassign? If so, I'd say this instruction can explicitly say it's "always initialized to" or something like that, since this instruction itself isn't capable of assigning another authority.

Reply:
works for me!